### DISCUSSION TITLE
How to understand async atoms?

### DISCUSSION THREAD

**Discussion by wong2** [pmndrs/jotai#2373] [General]
I've read the document https://jotai.org/docs/guides/async for several times and checked all the examples, but still can't understand how to use async atoms.

The example doesn't make sense to me:
```
const countAtom = atom(1)
const asyncAtom = atom(async (get) => get(countAtom) * 2)
```
Why would you read an atom value asynchronously? I think we need some real world examples to help understand.

Here is a pattern I always need in real app development:

- load initial value asynchronously
- update the value synchronously afterward

For example, load form values on mount, user update the values, then send the data on submission.
In code, I'm trying to do something like:

```
const formAtom = atom(async () => loadFormData())

// component
const [values, setValues] = useAtom(formAtom)

return (
  <Suspense>
    <form>
      <input name="email" value={values.email} onChange={() => setValues(....)} />
      ...
      <button type="submit">Submit</button>
    </form>
  </Suspense>
)
```

But of course it doesn't work.

**Comment by dai-shi** [comment]
Async atoms are for Suspense. If you don't need Suspense, it'd be an easier way.

That said, 

> - load initial value asynchronously
> - update the value synchronously afterward

is actually a difficult pattern. Maybe it's not documented well.
Here's a tweet: https://twitter.com/dai_shi/status/1714126922542964899

`atomWithDefault` might work too.
